docs/modules/ROOT/pages/explanations/commodore-components/kustomizations.adoc

```diff
@@ -0,0 +1,116 @@
+      namespace: ${slo:namespace}
+----
+
+In `component/fancy-sli-controller-overlay.jsonnet`, we render a kustomization overlay which refers to the kustomization and ensures the image as configured in `class/defaults.yml` is injected into the base.
+We also merge the provided `kustomize_input` into the overlay, to allow users to make arbitrary changes to the kustomization.
+
+.component/fancy-sli-controller-overlay.jsonnet
+[source,jsonnet]
+----
+// Template to render kustomization overlay
+local com = import 'lib/commodore.libjsonnet';
+local kap = import 'lib/kapitan.libjsonnet';
+
+local inv = kap.inventory();
+local params = inv.parameters.slo;
+
+local kustomization = {
+  kustomization: {
+    // Configure the provided kustomization as a base for our overlay
+    resources: [
+      '%s?ref=%s' % [
+        params.kustomization_url,
+        params.manifests_version,
+      ]
+    ],
+    // Inject the image configured in the component parameters
+    local image = params.images.fancy_sli_controller,
+    images: [ {
+      name: 'quay.io/syn/fancy-sli-controller',
+      newTag: image.tag,
+      newName: '%(registry)s/%(repository)s' % image,
+    } ],
+    // Inject the kustomize input provided in the component parameters
+  } + com.makeMergeable(params.kustomize_input),
+};
+
+{
+  // Render kustomization overlay as `kustomization.yaml`
+  kustomization: kustomization,
+}
+----
+
+In `class/slo.yml`, we render the kustomization overlay, and run `kustomize build` to generate the final Kubernetes resources.
+
+.class/slo.yml
+[source,yaml]
+----
+parameters:
+  kapitan:
+    inputs:
+      # render kustomization overlay into <component dir>/fancy-sli-controller
+      - input_paths:
+          - ${_base_directory}/component/fancy-sli-controller-overlay.jsonnet
+        input_type: jsonnet
+        output_path: ${_base_directory}/fancy-sli-controller
+      # Run kustomize on the rendered overlay.
+      # NOTE: Kapitan ignores `output_path` for `input_type=external`
+      - input_type: external
+        output_path: .
+        # Input path for `input_type=external` is the path to the external
+        # binary to call. Commodore provides parameter `_kustomize_wrapper`
+        # which expands to the absolute path of the kustomize wrapper script
+        # which is distributed with Commodore.
+        input_paths:
+          - ${_kustomize_wrapper}
+        env:
+          # The Commodore kustomize wrapper expects the input directory in
+          # environment variable INPUT_DIR.
+          INPUT_DIR: ${_base_directory}/fancy-sli-controller
+        # The Commodore kustomize wrapper always executes `kustomize build` and
+        # expects the first argument to be the output directory for `kustomize`.
+        # Users can provide arbitrary further arguments after the output
+        # directory which are passed to `kustomize` verbatim.
+        args:
+          - \${compiled_target_dir}/${_instance}/fancy-sli-controller-deployment
```

Comment:
What is up with the `\`? Is this escaping the `$`? But then why is the second one not escaped?

Reply:
See the updated version for an explanation for `\${compiled_target_dir}`